test/types/peg.test-d.ts

```diff
@@ -16,6 +15,15 @@ const src = readFileSync(
 
 const problems: peggy.Problem[] = [];
 
+function expectExact<T>() {
+  return function<U extends T>(
+    tValue: U
+  ): [T] extends [U] ? () => U : ["Argument of type", U, "does not match", T] {
+    // @ts-expect-error Just for type checking
+    return () => tValue;
+  };
+}
+
```

Comment:
On the one hand, I was inclined to keep the `expectType` name, but on the other, that would cause confusion for people familiar with `tsd`. But we could go either way.

We could also break this into two functions, which might make it easier to spot that you accidentally omitted the trailing `()`:

```
function exact<T>() {
   return function<U extends T>(
     tValue: U
   ): [T] extends [U] ? () => U : [U, "does not match", T] {
     // @ts-expect-error Just for type checking
     return () => tValue;
   };
 }
function check<T>(fn: ()=>T): void {
  fn();
}
```

And then call `check(exact<T>()(arg))`.

In fact, this makes the error message work better. Now its:
```
Argument of type '[string, "does not match", string | number]' is not assignable to parameter of type '() => unknown'.
  Type '[string, "does not match", string | number]' provides no match for the signature '(): unknown'.
```